contracts/TransmuteDPOS.sol

```diff
@@ -93,14 +84,41 @@ contract TransmuteDPOS is TransmuteToken, RoundManager, DelegatorManager {
   }
 
   function withdraw() external {
-    WithdrawInformation storage withdrawInformation = withdrawInformations[msg.sender];
-    require(withdrawInformation.withdrawBlock <= block.number);
-    require(delegatorStatus(msg.sender) == DelegatorStatus.UnbondedWithTokensToWithdraw);
-    this.transfer(msg.sender, withdrawInformation.amount);
-    delete withdrawInformations[msg.sender];
+    UnbondingInformation storage unbondingInformation = unbondingInformations[msg.sender];
+    require(unbondingInformation.withdrawBlock <= block.number);
+    require(delegatorStatus(msg.sender) == DelegatorStatus.Unbonding);
+    this.transfer(msg.sender, unbondingInformation.amount);
+    delete unbondingInformations[msg.sender];
+  }
+
+  function rebond(address _provider) external {
```

Comment:
why is rebond necessary, why not just have `bond` ?

Reply:
Good question, I did a first iteration bond() and rebond were merged into the same function but ran into problems if the amount you choose to rebond() is not the same the amount you bonded initially.

Example:
- D bond 10 TST to P
- D unbonds, so an unbondingInformation object is created containing "withdrawBlock" and "amount" information, that can be used for `withdraw()` or for `rebond()`
- if D rebonds less than 10 TST, let's say 6 there is a problem. It puts D in an in between state where he is Bonded to a new Provider but he still Unbonding because he has unbondingInformation (the 4 TST left). Moreover this new unbondingInformation object can't be treated like the previous one because he can only be `withdrawn()` not `rebonded()` (or else D would bond to two different providers which is against the rules).
- D rebonds exactly 10 TST then D is Bonded to the new Provider and unbondingInformation is deleted.

In short, it adds a lot less complexity if rebond() does not take an `amount` parameter and just automatically rebonds the full amount that has been unbonded.

Reply:
this makes sense.